Applications/Basic/CaptureHDRLoop/CaptureHDRLoop.cpp

```diff
@@ -0,0 +1,60 @@
+/*
+This example shows how to acquire HDR images from the Zivid camera in a loop
+(while actively changing some HDR settings).
+*/
+
+#include <Zivid/Zivid.h>
+
+#include <iostream>
+
+int main()
+{
+    try
+    {
+        Zivid::Application zivid;
+
+        std::cout << "Connecting to the camera" << std::endl;
+        auto camera = zivid.connectCamera();
+
+        std::cout << "Configuring the camera settings" << std::endl;
+        std::vector<std::vector<Zivid::Settings>> settingsHDRVector;
+
+        Zivid::Settings settings;
+        settings.set(Zivid::Settings::Brightness{ 1 })
+            .set(Zivid::Settings::Gain{ 1 })
+            .set(Zivid::Settings::Bidirectional{ false })
+            .set(Zivid::Settings::Filters::Contrast::Enabled::yes)
+            .set(Zivid::Settings::Filters::Contrast::Threshold{ 5 })
+            .set(Zivid::Settings::Filters::Gaussian::Enabled::yes)
+            .set(Zivid::Settings::Filters::Gaussian::Sigma{ 1.5 })
+            .set(Zivid::Settings::Filters::Outlier::Enabled::yes)
+            .set(Zivid::Settings::Filters::Outlier::Threshold{ 5 })
+            .set(Zivid::Settings::Filters::Reflection::Enabled::yes)
+            .set(Zivid::Settings::Filters::Saturated::Enabled::yes)
+            .set(Zivid::Settings::BlueBalance{ 1.081 })
+            .set(Zivid::Settings::RedBalance{ 1.709 });
+
+        std::vector<size_t> iris{ 10U, 20U, 30U };
+        std::vector<std::vector<int>> exposureTimeFrames{ { 10000, 90000, 40000 },
+                                                          { 40000, 10000, 90000 },
+                                                          { 90000, 40000, 10000 } };
+
+        for(size_t i = 0; i < 3; ++i)
+        {
+            std::vector<Zivid::Settings> settingsHDR;
+            for(size_t j = 0; j < 3; ++j)
+            {
+                settingsHDR.push_back(
+                    settings.set(Zivid::Settings::Iris{ iris[j] })
+                        .set(Zivid::Settings::ExposureTime{ std::chrono::microseconds{ exposureTimeFrames[i][j] } }));
```

Comment:
Shouldn't we use emplace_back? Do you know whether it's doing a copy, or move? I tried to read the push_back docs. Just worried we'll have three identical frames in each HDR.

Reply:
For `void push_back (const value_type& val);` it says that val is: value to be copied (or moved) to the new element.
For `void emplace_back (Args&&... args);` it says that args are: Arguments forwarded to construct the new element. This new element is constructed in place using args as the arguments for its constructor.

I think it doesn't matter which one we use, and is, in fact, the same in our case, as `settings` object already is created. It seems that it would matter only if the object didn't already exist and we wanted to pass just the arguments for creating settings object. Then we would have to use emplace_back.

@knatten can you help clarify this?

Reply:
@SatjaSivcev is correct, whenever you already have the full object, they are equivalent. `emplace_back` is faster if you only have the parts with which to construct the object.